indexer/services/comlink/src/controllers/api/v4/vault-controller.ts

```diff
@@ -470,6 +472,55 @@ function getPnlTicksWithCurrentTick(
   return pnlTicks.concat([currentTick]);
 }
 
+/**
+ * Takes in a map of block heights to PnlTicks and filters out the closest pnl tick per interval.
+ * @param pnlTicksByBlock
+ * @returns
+ */
+function filterOutIntervalTicks(
+  pnlTicksByBlock: Map<number, PnlTicksFromDatabase>,
+  resolution: PnlTickInterval,
+): PnlTicksFromDatabase[] {
+  // Track block to block time.
+  const blockToBlockTime: Map<number, DateTime> = new Map();
+  // Track start of days to closest block by block time.
+  const blocksPerInterval: Map<string, number> = new Map();
+  // Track start of days to closest Pnl tick.
+  const ticksPerInterval: Map<string, PnlTicksFromDatabase> = new Map();
+  pnlTicksByBlock.forEach((pnlTick: PnlTicksFromDatabase, block: number): void => {
+    const blockTime: DateTime = DateTime.fromISO(pnlTick.blockTime).toUTC();
+    blockToBlockTime.set(block, blockTime);
+
+    const startOfInterval: DateTime = blockTime.toUTC().startOf(resolution);
+    const startOfIntervalBlock: number | undefined = blocksPerInterval.get(startOfInterval.toISO());
```

Comment:
nit: not sure how much work this call is. if notable, define `startOfInterval.toISO()` as a const so we don't call it many times?